designer/server/src/test/scala/pl/touk/nussknacker/ui/integration/DictsFlowTest.scala

```diff
@@ -114,6 +117,164 @@ class DictsFlowTest
+
+    val processName = "test_proc"
+
+    createEmptyScenario(ProcessName(processName))
+
+    val process = ScenarioBuilder
+      .streaming(processName)
+      .additionalFields(properties = Map.empty)
+      .source("source", "csv-source-lite")
+      .fragment(
+        fragmentName,
+        fragmentName,
+        List(("test", Expression.dictKeyWithLabel("H000000", None))),
+        Map.empty,
+        Map.empty
+      )
+
+    val globalErrorProcess = extractGlobalValidationResult(process)
+    globalErrorProcess.map(_.error.typ) shouldBe List("InvalidTailOfBranch")
+    globalErrorProcess.map(_.nodeIds) shouldBe List(List("test_frag"))
+
+    // Check that label bound to dictId-key is updated in BE response
+    val response2 = httpClient.send(
+      quickRequest
+        .get(uri"$nuDesignerHttpAddress/api/processes/$processName")
+        .auth
+        .basic("admin", "admin")
+    )
+    response2.code shouldEqual StatusCode.Ok
+
+    response2.bodyAsJson.hcursor
+      .downField("scenarioGraph")
+      .downField("nodes")
+      .downAt(_.hcursor.get[String]("id").rightValue == "test_frag")
+      .downField("ref")
+      .downField("parameters")
+      .downAt(_.hcursor.get[String]("name").rightValue == "test")
+      .downField("expression")
+      .downField("expression")
+      .as[String]
+      .rightValue shouldBe """{"key":"H000000","label":"Black"}"""
+  }
+
+  test("handle labels correctly even if used fragment's resolution fails - unused output") {
+    val fragmentName = "test_frag"
+
+    createEmptyScenario(ProcessName(fragmentName), isFragment = true)
+
+    val fragment = ScenarioBuilder
+      .fragment(
+        fragmentName,
+        List(
+          FragmentParameter(
+            ParameterName("test"),
+            FragmentClazzRef[java.lang.String],
+            initialValue = None,
+            hintText = None,
+            valueEditor = Some(
+              ValueInputWithDictEditor(
+                dictId = "rgb",
+                allowOtherValue = false
+              )
+            ),
+            valueCompileTimeValidation = None
+          )
+        )
+      )
+      .fragmentOutput("output", "output")
+
+    extractNodeValidationResult(fragment).asObject.value shouldBe empty
+
+    val processName = "test_proc"
+
+    createEmptyScenario(ProcessName(processName))
+
+    // This process uses the defined fragment (which has output - not a sink), but doesn't direct it's output anywhere
+    val process = ScenarioBuilder
+      .streaming(processName)
+      .additionalFields(properties = Map.empty)
+      .source("source", "csv-source-lite")
+      .fragment(
+        fragmentName,
+        fragmentName,
+        List(("test", Expression.dictKeyWithLabel("H000000", None))),
+        Map("output" -> "fragmentOut"),
+        Map.empty
+      )
+
+    val globalErrorProcess = extractGlobalValidationResult(process)
+    globalErrorProcess.map(_.error.typ) shouldBe List("InvalidTailOfBranch")
```

Comment:
This error type looks hacky in this context. I'd rather expect sth like output of fragment is not used or sth similar. Is it easy to change? Can it be node error instead of global error?